public/js/notifications.js

```diff
@@ -0,0 +1,433 @@
+                if (this.eventSource.readyState !== EventSource.CLOSED) {
+                    this.eventSource.close();
+                }
+            }
+            this.eventSource = null;
+            this.initialized = false;
+
+            this.logger.debug(this.prefix + "unloaded");
+        }
+
+        reload () {
+            this.unload();
+            this.load();
+        }
+
+        openEventStream () {
+            if (this.hasNotificationPermission() && this.hasNotificationsEnabled()) {
+                // close existing event source object if there's one
+                if (this.eventSource !== null && this.eventSource.readyState !== EventSource.CLOSED) {
+                    this.eventSource.close();
+                }
+
+                try {
+                    this.logger.debug(this.prefix + "opening event source");
+                    this.eventSource = new EventSource(icinga.config.baseUrl + '/notifications/daemon', {withCredentials: true});
+                    this.eventSource.addEventListener('message', () => {
+                        // this._logger.debug(this._prefix + `got message from event-stream: ${event.data}`);
+                    });
+                    this.eventSource.addEventListener('error', () => {
+                        // this._logger.debug(this._prefix +`got an error from event-stream: ${event}`);
+                    });
+                    this.eventSource.addEventListener('icinga2.notification', (event) => {
+                        if (this.hasNotificationPermission() && this.hasNotificationsEnabled()) {
+                            // send to service_worker if the permissions are given and the notifications enabled
+                            this.getServiceWorker()
+                                .then((serviceWorker) => {
+                                    if (serviceWorker) {
+                                        serviceWorker.postMessage(
+                                            JSON.stringify({
+                                                command: 'notification',
+                                                notification: JSON.parse(event.data),
+                                                baseUrl: icinga.config.baseUrl
+                                            })
+                                        );
+                                    }
+                                });
+                        }
+                    });
+                } catch (error) {
+                    // this._logger.error(this._prefix + `got an error: ${error}`);
+                }
+            } else {
+                this.logger.warn(this.prefix + "denied opening event-stream as the notification permissions" +
+                    " are missing or the notifications themselves disabled");
+            }
+        }
+
+        renderHandler (event) {
+            if (event.type === 'rendered') {
```

Comment:
This condition must be superfluous, as the event type must always be the `rendered`.